changelog/13737.feature.rst

```diff
@@ -0,0 +1,3 @@
+Added flag --require-unique-paramset-ids to pytest that when set to true will
+make pytest raise an exception when non-unique
+paramset ids are detected instead of attempting to generate them
```

Comment:
```suggestion
Added the ``--require-unique-paramset-ids`` flag to pytest. 

When passed, this flag causes pytest to raise an exception upon detection of non-unique parameter set IDs, rather than attempting to generate them automatically.
```

Reply:
Could we enrich this entry with examples where this is useful? The issue mentions "for example, when plugins are doing fancy collection tricks and then a subsequent run tries to select a subset of those nodeids", something like that.